durable-storage/src/merkle_layer/node_wrapper.rs

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: 2025 TriliTech <contact@trili.tech>
+// SPDX-License-Identifier: MIT
+
+use std::sync::Arc;
+use std::sync::OnceLock;
+
+use crate::persistence_layer::PersistenceLayer;
+
+use super::node::MavlNode;
+use super::node::MavlNodeHashRepresentation;
+use octez_riscv_data::serialisation::borrow_decode;
+
+#[derive(Debug, Clone, Default)]
+pub(super) struct NodeWrapper {
+    pub node: OnceLock<Option<Arc<MavlNode>>>,
+    commited_hash: Option<blake3::Hash>,
+    persistence_layer: Arc<OnceLock<PersistenceLayer>>,
```

Comment:
How do you support swapping the `PersistenceLayer` without cloning, when an `MavlNode` contains this `NodeWrapper` but needs to be paired with a new `PersistenceLayer`. For example, when the database gets cloned, as that creates a new Persistence layer.